code.js

```diff
@@ -1,9 +1,11 @@
 function factorial(n) {
-    if(n === 0) return 1;
-    else return n * factorial(n - 1);
-}
-
-function e(n) {
-    if(n === 0) return 1;
-    else return 1.0 / factorial(n) + e(n - 1);
+    let sum = 1;
```

Comment:
Dynamic Programming uses a recurrent formula to find the solution. It is similar to the divide and conquer.